packages/flutter/lib/src/cupertino/button.dart

```diff
@@ -0,0 +1,167 @@
+    color: kCupertinoWhite,
+  );
+
+  static const EdgeInsets _kCupertinoButtonPadding = const EdgeInsets.all(16.0);
+  static const EdgeInsets _kCupertinoBackgroundButtonPadding =
+      const EdgeInsets.symmetric(vertical: 16.0, horizontal: 64.0);
+
+  CupertinoButton({
+    @required this.child,
+    this.padding,
+    this.color,
+    @required this.onPressed,
+  });
+
+  final Widget child;
+
+  /// The amount of space to surround the child inside the bounds of the button.
+  ///
+  /// Defaults to 16.0 pixels.
+  final EdgeInsets padding;
+
+  /// The color of the button's background.
+  ///
+  /// Defaults to null which produces a button with no background or border.
+  final Color color;
+
+  /// The callback that is called when the button is tapped or otherwise activated.
+  ///
+  /// If this is set to null, the button will be disabled.
+  final VoidCallback onPressed;
+
+  /// Whether the button is enabled or disabled. Buttons are disabled by default. To
+  /// enable a button, set its [onPressed] property to a non-null value.
+  bool get enabled => onPressed != null;
+
+  @override
+  _CupertinoButtonState createState() => new _CupertinoButtonState();
+
+  @override
+  void debugFillDescription(List<String> description) {
+    super.debugFillDescription(description);
+    if (!enabled)
+      description.add('disabled');
+  }
+}
+
+class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProviderStateMixin {
+  static const Duration kFadeOutDuration = const Duration(milliseconds: 10);
+  static const Duration kFadeInDuration = const Duration(milliseconds: 350);
+
+  AnimationController _animationController;
+
+  @override
+  void initState() {
+    super.initState();
+    _animationController = new AnimationController(
+      duration: const Duration(milliseconds: 200),
+      value: 1.0,
+      vsync: this,
+    );
+  }
+
+  @override
+  void dispose() {
+    _animationController.dispose();
+    _animationController = null;
+    super.dispose();
+  }
+
+  void _handleTapDown(PointerDownEvent event) {
+    _animationController.animateTo(0.1, duration: kFadeOutDuration);
+  }
+
+  void _handleTapUp(PointerUpEvent event) {
+    _animationController.animateTo(1.0, duration: kFadeInDuration);
+  }
+
+  void _handleTapCancel(PointerCancelEvent event) {
+    _animationController.animateTo(1.0, duration: kFadeInDuration);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final bool enabled = config.enabled;
+    final Color backgroundColor = config.color;
+
+    return new Listener(
+      onPointerDown: enabled ? _handleTapDown : null,
+      onPointerUp: enabled ? _handleTapUp : null,
+      onPointerCancel: enabled ? _handleTapCancel : null,
+
+      child: new GestureDetector(
+        onTap: config.onPressed,
+        child: new ConstrainedBox(
+          constraints: const BoxConstraints(minWidth: 48.0, minHeight: 48.0),
```

Comment:
If you just put this widget in a Center inside a Scaffold body, I think you'll end up with something super tall.  You might want to set the maxHeight as well.